### PR TITLE
test: extend raw connection driver

### PR DESCRIPTION
**Pull request by lambdai**
Commit Message:
Avoid dispatcher allocation. Allow passing ssl transport socket. Add waitForConnected helper method.
Additional Description:
Inspired and will be used by https://github.com/envoyproxy/envoy/pull/10662
Risk Level: LOW
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/integration/utility.cc [hunk 1]
```diff
@@ -114,15 +114,22 @@ IntegrationUtil::makeSingleRequest(uint32_t port, const std::string& method, con
 
 RawConnectionDriver::RawConnectionDriver(uint32_t port, Buffer::Instance& initial_data,
                                          ReadCallback data_callback,
-                                         Network::Address::IpVersion version) {
+                                         Network::Address::IpVersion version,
+                                         Event::Dispatcher& dispatcher,
```
Comment (mattklein123) [inline test/integration/utility.cc:118]: Would it be better to have a method on BaseIntegrationTest that created a raw connection driver and implicitly passed the test dispatcher? Might be less change?

Comment (lambdai) [reply test/integration/utility.cc:118]: Great idea! Added that helper and use it everywhere

**Comment by mattklein123** [comment]
Nice! Can you check CI?

/wait

**Comment by lambdai** [comment]
It's a build failure at https://github.com/envoyproxy/envoy-filter-example because the RawConnectionDriver ctor update.

It's a chicken-egg problem. One of the solution is to migrate in 3 steps
1. provide both new and old interface in envoy, land the PR
2. when (1) is done, update envoy-filter-example, switching to new interface, land the PR
3. when (2) is done, delete the old interface in envoy

Any alternative? @mattklein123

**Comment by mattklein123** [comment]
You can make the update in filter-example and we can force merge it, and then you can include that SHA update in this PR (we pin to a SHA of filter example in do_ci.sh)

**Comment by lambdai** [comment]
Create https://github.com/envoyproxy/envoy-filter-example/pull/122

**Pull request merged**